tests/rules/taint_exception.py

```diff
@@ -0,0 +1,242 @@
+# This test aims to exercise various exception paths.
```

Comment:
It's also interesting to write a test using constant propagation, because it will also be affected by this change, and it behaves differently than taint (must vs may).

Reply:
Sounds good! Next update will include some constant propagation tests.